src/core_modules/capture-core/components/WidgetEnrollment/TransferModal/OrgUnitField/useSearchScopeWithFallback.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+
+import { useApiMetadataQuery } from '../../../../utils/reactQueryHelpers';
+
+type Props = {
+    searchText: ?string,
+};
+
+export const useSearchScopeWithFallback = ({ searchText }: Props) => {
+    const { data: orgUnitRoots, isLoading } = useApiMetadataQuery(
+        ['organisationUnits', 'userOrgUnitScope'],
+        {
+            resource: 'me',
+            params: {
+                fields: 'teiSearchOrganisationUnits[id,path],organisationUnits[id,path]',
+            },
+        },
+        {
+            enabled: !searchText,
+            select: (data) => {
+                const { teiSearchOrganisationUnits, organisationUnits } = data;
+                return teiSearchOrganisationUnits.length
+                    ? teiSearchOrganisationUnits
+                    : organisationUnits;
+            },
+        },
+    );
+
+    const { data: searchOrgUnits, isLoading: isLoadingSearch } = useApiMetadataQuery(
+        // $FlowFixMe - react-query types are not up to date
+        ['organisationUnits', 'userOrgUnitScope', 'search', searchText],
```

Comment:
I'm not sure what you meant here. I think in general that orgunits are considered so stable that it would not change during a user session. We could limit cacheTime, but not sure if I understand the reasoning behind it. Could you elaborate?

Reply:
I just think it makes little sense to permanently store the result of a query based on a "semi-random" string, because it is probably unlikely that the exact same query string will be used again during the same session. But it's certainly not impossible, so a cache time of 1 hr maybe to keep the most frequent queries?